R/ggplotly.R

```diff
@@ -702,6 +702,7 @@ gg2list <- function(p, width = NULL, height = NULL, tooltip = "all",
   )
   
   # if theme(legend.position = "none") is used, don't show a legend _or_ guide
+  npscales$scales <- Filter(function(x) x$guide != "none", npscales$scales)
   if (npscales$n() == 0 || identical(theme$legend.position, "none")) {
```

Comment:
I see this as potentially dangerous -- a scale without a guide doesn't necessarily mean it doesn't exist. 

I would prefer to instead set `layout.showlegend` to `FALSE` for discrete scales and remove the colorbar for continuous ones.